src/cmd/serial.js

```diff
@@ -233,133 +243,308 @@ module.exports = class SerialCommand extends CLICommandBase {
 	}
 
 	/**
-	 * Check to see if the device is in listening mode, try to get the device ID via serial
+	 * Check to see if the device is in listening mode
+	 * Device identify gives the following
+	 *     - (Wifi and Cellular devices) Device ID
+	 *     - (Cellular devices) Cell radio IMEI
+	 *           - Obtained via control req for dvos > 5.6.0
+	 *           - Obtained via serial otherwise
+	 *     - (Cellular devices) Cell radio ICCID
+	 *     - (Wifi and Cellular devices) System firmware version
 	 * @param {Number|String} comPort
 	 */
-	identifyDevice({ port }){
+	async identifyDevice({ port }){
 		let device;
-		return this.whatSerialPortDidYouMean(port, true)
-			.then(_device => {
-				device = _device;
-				if (!device){
-					throw new VError('No serial port identified');
+		const deviceFromSerialPort = await this.whatSerialPortDidYouMean(port, true);
+
+		if (!deviceFromSerialPort || !deviceFromSerialPort.deviceId) {
+			throw new VError('No serial port identified');
+		}
+
+		try {
+			const deviceId = deviceFromSerialPort.deviceId;
+
+			device = await usbUtils.getOneUsbDevice({ idOrName: deviceId });
+
+			const fwVer = await device.getSystemVersion({ timeout: 2000 });
+
+			console.log();
+			console.log('Your device id is', chalk.bold.cyan(deviceId));
+
+			const platform = deviceFromSerialPort.specs.name;
+			const features = deviceConstants[platform].features;
+
+			if (features.includes('cellular')) {
+				// Add checking for new dvos versions
+				let res;
+				if (semver.lt(fwVer, '5.6.0')) { // FIXME: Get the correct firmware version
+					res = await this._identifyCellularDeviceOverSerial(deviceFromSerialPort);
+				} else {
+					res = await this._identifyCellularDeviceOverCtrlReq(device);
 				}
```

Comment:
I'd prefer not to keep a part of the identify command using a serial one letter request. I'd rather omit information that is not available as long as it's not critical. In this case, I expect nobody cares about the IMEI.